src/Host/Client/Impl/Definitions/IRSessionProvider.cs

```diff
@@ -5,19 +5,28 @@
 using System.Collections.Generic;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.R.Host.Client.Host;
 
 namespace Microsoft.R.Host.Client {
     public interface IRSessionProvider : IDisposable {
-        IRSession GetOrCreate(Guid guid);
+        IRSession GetOrCreate(Guid guid, IRHostBrokerConnector brokerConnector);
```

Comment:
You don't usually talk to the broker or session provider. There are just 8 usages of the `IRSessionProvider` interface in the whole product code - the rest are in test code. We will have to decide how secondary sessions should work when user has more than one REPL (i.e., what should happen with intellisense when user switches from one REPL to another)

---
In reply to: [73035733](https://github.com/Microsoft/RTVS/pull/2133#discussion_r73035733) [](ancestors = 73035733)